Source/Fuse.Controls.Video/Android/VideoPlayer.uno

```diff
@@ -196,25 +197,59 @@ namespace Fuse.Controls.VideoImpl.Android
 
 		readonly VideoTexture _videoTexture;
 
+		string _dataSourcePath;
+
 		public MediaPlayer()
 		{
 			var glHandle = GL.CreateTexture();
 			_videoTexture = new VideoTexture(glHandle);
 			_surfaceTexture = CreateSurfaceTexture((int)glHandle);
 			_surface = CreateSurface(_surfaceTexture);
 			_handle = CreateMediaPlayer(_surface);
+			_dataSourcePath = null;
 
 			Fuse.Platform.Lifecycle.EnteringBackground += OnEnteringBackground;
 		}
 
 		[Foreign(Language.Java)]
-		static int GetOrientation(Java.Object handle)
+		static int GetOrientation(Java.Object handle, string dataSorucePath)
 		@{
+			/*
+				Nasty code to check for rotation metadata on a video
+			*/
+			if (dataSorucePath != null)
+			{
+				try
+				{
+					android.media.MediaMetadataRetriever mmr = new android.media.MediaMetadataRetriever();
+					mmr.setDataSource(dataSorucePath);
+					String rotation = mmr.extractMetadata(android.media.MediaMetadataRetriever.METADATA_KEY_VIDEO_ROTATION);
+					if (rotation != null) {
+						return java.lang.Integer.parseInt(rotation);
+					}
+				}
+				catch(Exception e) {}
```

Comment:
I'd prefer to see either a diagnostic here, or a comment indicating why we could just ignore the exception.